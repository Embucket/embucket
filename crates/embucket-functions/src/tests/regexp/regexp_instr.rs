use crate::test_query;

test_query!(
    regexp_instr_basic,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_position,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d', 5)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_occurrence,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d', 1, 3)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_option_start,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d', 1, 1, 0)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_option_end,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d', 1, 1, 1)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_occurence_limit,
    "SELECT REGEXP_INSTR('nevermore1, nevermore2, nevermore3.', 'nevermore\\d', 1, 4)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_capture_groups,
    "SELECT REGEXP_INSTR(column1, 'the\\W+\\w+')
    FROM VALUES ('It was the best of times, it was the worst of times.'),
    ('In    the   string   the   extra   spaces  are   redundant.'),
    ('A thespian theater is nearby.')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_capture_groups_occurrence,
    "SELECT REGEXP_INSTR(column1, 'the\\W+\\w+', 1, 2)
    FROM VALUES ('It was the best of times, it was the worst of times.'),
    ('In    the   string   the   extra   spaces  are   redundant.'),
    ('A thespian theater is nearby.')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_capture_groups_group_num,
    "SELECT REGEXP_INSTR(column1, 'the\\W+(\\w+)', 1, 2, 0, 'e', 1)
    FROM VALUES ('It was the best of times, it was the worst of times.'),
    ('In    the   string   the   extra   spaces  are   redundant.'),
    ('A thespian theater is nearby.')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_capture_groups_regexp_parameter_no_group_num,
    "SELECT REGEXP_INSTR(column1, 'the\\W+(\\w+)', 1, 2, 0, 'e')
    FROM VALUES ('It was the best of times, it was the worst of times.'),
    ('In    the   string   the   extra   spaces  are   redundant.'),
    ('A thespian theater is nearby.')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_capture_groups_group_num_no_regexp_parameter,
    "SELECT REGEXP_INSTR(column1, 'the\\W+(\\w+)', 1, 2, 0, '', 1)
    FROM VALUES ('It was the best of times, it was the worst of times.'),
    ('In    the   string   the   extra   spaces  are   redundant.'),
    ('A thespian theater is nearby.')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_word_groups,
    "SELECT REGEXP_INSTR(column1, 'A\\W+(\\w+)', 1, 1, 0, 'e', 1),
    REGEXP_INSTR(column1, 'A\\W+(\\w+)', 1, 2, 0, 'e', 1),
    REGEXP_INSTR(column1, 'A\\W+(\\w+)', 1, 3, 0, 'e', 1),
    REGEXP_INSTR(column1, 'A\\W+(\\w+)', 1, 4, 0, 'e', 1)
    FROM VALUES ('A MAN A PLAN A CANAL')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_letter_groups,
    "SELECT REGEXP_INSTR(column1, 'A\\W+(\\w)(\\w)(\\w)', 1, 1, 0, 'e', 1),
    REGEXP_INSTR(column1, 'A\\W+(\\w)(\\w)(\\w)', 1, 1, 0, 'e', 2),
    REGEXP_INSTR(column1, 'A\\W+(\\w)(\\w)(\\w)', 1, 1, 0, 'e', 3)
    FROM VALUES ('A MAN A PLAN A CANAL')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_word_boundary,
    "SELECT REGEXP_INSTR('It was the best of times, it was the worst of times','\\bwas\\b', 1, 1)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_regex_patterns_1,
    "SELECT REGEXP_INSTR('It was the best of times, it was the worst of times', '[[:alpha:]]{2,}st', 15, 1)",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_regex_patterns_2,
    "SELECT REGEXP_INSTR(column1, '\\b\\S*o\\S*\\b')
    FROM VALUES ('Hellooo World'),
    ('How are you doing today?'),
    ('the quick brown fox jumps over the lazy dog'),
    ('PACK MY BOX WITH FIVE DOZEN LIQUOR JUGS')",
    snapshot_path = "regexp_instr"
);

test_query!(
    regexp_instr_regex_patterns_3,
    "SELECT REGEXP_INSTR(column1, '\\b\\S*o\\S*\\b', 3, 3, 1, 'i')
    FROM VALUES ('Hellooo World'),
    ('How are you doing today?'),
    ('the quick brown fox jumps over the lazy dog'),
    ('PACK MY BOX WITH FIVE DOZEN LIQUOR JUGS')",
    snapshot_path = "regexp_instr"
);
