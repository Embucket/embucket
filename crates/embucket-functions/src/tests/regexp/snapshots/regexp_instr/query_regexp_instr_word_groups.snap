---
source: crates/embucket-functions/src/tests/regexp/regexp_instr.rs
description: "\"SELECT REGEXP_INSTR(column1, 'A\\\\W+(\\\\w+)', 1, 1, 0, 'e', 1),\n    REGEXP_INSTR(column1, 'A\\\\W+(\\\\w+)', 1, 2, 0, 'e', 1),\n    REGEXP_INSTR(column1, 'A\\\\W+(\\\\w+)', 1, 3, 0, 'e', 1),\n    REGEXP_INSTR(column1, 'A\\\\W+(\\\\w+)', 1, 4, 0, 'e', 1)\n    FROM VALUES ('A MAN A PLAN A CANAL')\""
---
Ok(
    [
        "+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+",
        "| regexp_instr(column1,Utf8(\"A\\W+(\\w+)\"),Int64(1),Int64(1),Int64(0),Utf8(\"e\"),Int64(1)) | regexp_instr(column1,Utf8(\"A\\W+(\\w+)\"),Int64(1),Int64(2),Int64(0),Utf8(\"e\"),Int64(1)) | regexp_instr(column1,Utf8(\"A\\W+(\\w+)\"),Int64(1),Int64(3),Int64(0),Utf8(\"e\"),Int64(1)) | regexp_instr(column1,Utf8(\"A\\W+(\\w+)\"),Int64(1),Int64(4),Int64(0),Utf8(\"e\"),Int64(1)) |",
        "+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+",
        "| 3                                                                                     | 9                                                                                     | 16                                                                                    | 0                                                                                     |",
        "+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------+",
    ],
)
